### PR TITLE
Translation (zh-hant.php)

### PR DESCRIPTION
**Pull request by Taiwania**
Full translated.

### CODE REVIEW

**Comment by Taiwania** [comment]
......It's my first time to use GitHub. If I'm doing something wrong, please talk to me.

**Comment by Taiwania** [comment]
Fixed: step 5/5 translated.

File: i18n/zh-hant.php [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+              if (!$("#s4fg").hasClass("has-error")) {
+                  $("#s4fg").addClass("has-error");
+              }
+          } else {
+              $("form").submit();
+          }
+      }
+    </script>
+  </head>
+  <body style="background: url('bg.png') no-repeat fixed right bottom; overflow:hidden;">
```
Comment (Artoria2e5) [inline i18n/zh-hant.php:102]: Asking for some `lang="zh-Hant"` here.

File: i18n/zh-hant.php [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+                    scrollTop: $(result).offset().top
+                }, 400);
+            });
+            </script>
+            <?php
+        }
+    ?>
+
+    <div class="container">
+      <h1>維基媒體 OTRS 釋出產生器 <small><a id="meta" tabindex="0" data-toggle="popover" data-placement="bottom" data-trigger="focus" data-content="created and maintained by <a href='//meta.wikimedia.org/wiki/User:FDMS4' target='_blank'>FDMS</a><br />© (<a href='//joinup.ec.europa.eu/sites/default/files/eupl1.1.-licence-en_0.pdf' target='_blank'>EUPL 1.1</a>) <a href='http://fdms.eu' target='_blank'>Stöger Florian D. M.</a" style="color:#777;"><?php echo $relgen;?></a></small></h1>
```
Comment (Artoria2e5) [inline i18n/zh-hant.php:144]: Use CJK parentheses. (after &copy;)

Comment (Artoria2e5) [reply i18n/zh-hant.php:144]: Focus text (created and maintained by) not translated.

Comment (Artoria2e5) [reply i18n/zh-hant.php:144]: Just out of curiosity, why all these `//` protocol-relative links? WMF is using HTTPS anyway...

File: i18n/zh-hant.php [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+                <br />
+                著作權持有者<br /><div id="s1fg2" class="form-group"><input id="repi" type="text" name="rep" value="<?php echo $rep;?>" placeholder="王牌企業 / 李春嬌 (必填)" class="form-control" /></div>
+                <br />
+                我的職權<br /><div id="s1fg3" class="form-group"><input id="authi" type="text" name="auth" value="<?php echo $auth;?>" placeholder="執行長、指定代表等 (必填)" class="form-control" /></div>
+              </div>
+              <br /><br />
+              <a role="button" class="btn btn-primary btn-block nt" onclick="s1v()">前進至下一步</a>
+            </div>
+            <div style="display:none;" id="idk">
+              <p class="text-danger">維基媒體 OTRS 不能接受您的釋出授權──請向<a href="//commons.wikimedia.org/wiki/Commons:OTRS#notch">該作品的著作權持有人聯繫</a>。</p>
```
Comment (Artoria2e5) [inline i18n/zh-hant.php:184]: Use two U+2014's (——) for Chinese em dash. See Wikipedia MOS and W3C CLREQ.

File: i18n/zh-hant.php [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+          <br /><br />
+          <div class="col-md-7">
+            <div class="button" data-toggle="buttons">
+              <label class="btn btn-default btn-block" onclick="$('#i3').show();"><input type="radio" id="s31" name="s3" value="1" />我要釋出媒體作品</label>
+              <label class="btn btn-default btn-block" onclick="$('#i3').show();"><input type="radio" id="s32" name="s3" value="2" />我要釋出從某個媒體作品當中所創作的作品</label>
+              <label class="btn btn-default btn-block" onclick="$('#i3').show();"><input type="radio" id="s33" name="s3" value="3" />我要釋出媒體作品與從其作品當中創作的作品</label>
+            </div>
+            <br />
+            <p>如果您的作品是從某人的作品以複雜方式進行創作或將某人的作品包含其中，那麼該作品就是一件<a href="//commons.wikimedia.org/wiki/Commons:Derivative_works" target="_blank"><b>衍生創作作品</b></a>，一般必須要提供該作品與其原始作品的著作權持有者釋出許可。</p>
+            <p>在部分國家，由於有<a href="//commons.wikimedia.org/wiki/Commons:Freedom_of_panorama" target="_blank"><b>全景自由攝影</b></a>的例外，建築物或其他永久公開展示的作品可不用提供原始著作權持有者釋出許可。在美國，只有建築物不需要提供原始著作權持有者的許可。</p>
```
Comment (Artoria2e5) [inline i18n/zh-hant.php:228]: Disputing the interpretation of panorama, and the phrasing of the whole world here. Try "室外公共場合攝影之自由". (Freedom of photography in public outdoor spaces.)

Well yeah try something that includes paintings. Hmm, "攝影、繪圖"? "製圖"?

Comment (Artoria2e5) [reply i18n/zh-hant.php:228]: See also https://commons.wikimedia.org/wiki/Commons:Freedom_of_panorama#Taiwan_.28Republic_of_China.29. You may want to add a note on this painting-only stuff.

File: i18n/zh-hant.php [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+          </div>
+          <br />
+        </div>
+
+        <div id="s4" class="row hof"> <!-- step 4 -->
+          <br />
+          步驟 4/5 [ <a href="#s3" class="smsc">返回</a> | <a href="//commons.wikimedia.org/wiki/Commons:Help_desk?action=edit&section=new&preloadtitle=help+with+Wikimedia+OTRS+release+generator+step+4" target="_blank">說明</a> | <a href="//commons.wikimedia.org/wiki/User_talk:FDMS4?action=edit&section=new&preloadtitle=Wikimedia+OTRS+release+generator+feedback" target="_blank">回饋</a> ]
+          <br /><br />
+          <div class="col-md-7">
+            <p>我同意將上述提到的內容以下列自由授權釋出：</p>
```
Comment (Artoria2e5) [inline i18n/zh-hant.php:244]: My `zh-cmn` parser is parsing `授權` (license) as a verb modified by `自由` (freedom). Try `授權書`.

File: i18n/zh-hant.php [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+            <div id="s4fg" class="form-group"><div class="input-group">
+              <input id="licensei" type="text" name="license" value="創用CC 姓名標示-相同方式分享 4.0 國際版" class="form-control" />
+              <div class="input-group-btn">
+                <a role="button" data-toggle="dropdown" class="btn btn-default"><span class="caret" /></a>
+                <ul class="dropdown-menu dropdown-menu-right">
+                  <li><a onclick="$('#licensei').val('創用CC 姓名標示-相同方式分享 4.0 國際版'); $('#iawattr').show();">創用CC 姓名標示-相同方式分享 4.0 國際版</a></li>
+                  <li><a onclick="$('#licensei').val('創用CC 姓名標示 4.0 國際版'); $('#iawattr').show();">創用CC 姓名標示 4.0 國際版</a></li>
+                  <li><a onclick="$('#licensei').val('創用CC CC0 1.0 通用版'); $('#iawattr').hide();">創用CC CC0 1.0 通用版 (貢獻至公共領域)</a></li>
```
Comment (Artoria2e5) [inline i18n/zh-hant.php:252]: Parentheses. Not gonna label them all, so look for these yourself.

File: i18n/zh-hant.php [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+          fputcsv($stats, array (date("m-d"), $starttime, date("H:i:s"), $trn), ";");
+          fclose($stats);
+        ?>
+        <div class="col-md-7">
+          <p>如果您有安裝電子郵件用戶端，只要<b>點選右側的按鈕</b>，就會建立一封釋出授權電子郵件。如果沒有安裝 (或點選過後沒有反應)，請手動複製下方綠色方塊中的文字，貼到您的電子郵件中，寄到 <a href="mailto:permissions-zh@wikimedia.org">permissions-zh@wikimedia.org</a>。</p>
+          <p>這封電子郵件應當要從<b>與其釋出作品內容有相關並可以辨識出來的電子郵件位址</b>寄送出去。舉例，如果您所釋出的圖像於某個網站中展示，您的電子郵件位址應當與該網站或列於該網站的聯絡資訊頁面有所關連；如果您釋出的圖像代表某個組織，您的電子郵件位址應當為該組織的官方電子郵件位址。</p>
+          <br />
+          <?php
+            if ($s1 == "1") {
+                $p1s = "$name 是";
```
Comment (Artoria2e5) [inline i18n/zh-hant.php:291]: Disputing the space placement here. People using a `zh` form likely give Chinese names, therefore spaces should not be needed.

File: i18n/zh-hant.php [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+          ?>
+          <br /><br />
+        </div>
+        <div class="col-md-4">
+          <a role="button" href="mailto:permissions-zh@wikimedia.org?subject=<?=$subj?>&amp;body=<?=$b1m?>%0A<?=$b2?>%0A<?=$b3?>%0A<?=$b4?><?=$b5m?>%0A<?=$b6?>%0A%0A<?=$name?><?=$p1s_m?>%0A<?=date('Y-m-d')?>%0A%0A<?=$tracking?>" class="btn btn-default btn-block" style="width:100%;">建立釋出電子郵件</a>
+        </div>
+        <?php
+            } else {
+                if ($name == "") echo "<p class='text-danger'>錯誤：沒有填入姓名！</p>";
+                if (($s1 == "2") && (($rep == "") || ($auth == ""))) echo "<p class='text-danger'>錯誤：沒有填入著作權持有者與/或職權！</p>";
```
Comment (Artoria2e5) [inline i18n/zh-hant.php:338]: I believe taking the `or` (`或`) is good enough.

File: i18n/zh-hant.php [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+              $("form").submit();
+          }
+      }
+    </script>
+  </head>
+  <body style="background: url('bg.png') no-repeat fixed right bottom; overflow:hidden;">
+
+    <?php
+        $relgen = "0.9.9";
+        date_default_timezone_set("UTC");
```
Comment (Artoria2e5) [inline i18n/zh-hant.php:106]: Is this worthy of localization by the way?

**Review by Artoria2e5 — CHANGES_REQUESTED** [review]
Almost good.

File: i18n/zh-hant.php (hunk 2 above)
Comment (Artoria2e5) [reply i18n/zh-hant.php:144]: Try using `釋出書`/`釋出信` to show readers that you are talking about a letter.

Comment (Sekidoki) [reply i18n/zh-hant.php:144]: Use `釋出授權信` maybe better. Prove it later.

**Pull request merged**